### ISSUE TITLE
Appliance upload stuck at 2% (13 MB)

### ISSUE THREAD

**Issue by ventaquil** [GNS3/gns3-server#2310]
I am using the newest version (3.0.0a5) of GUI and server.

I am running GUI on Windows 10 and server on Ubuntu 20.04 - remote server was configured as main and only server.

During upload of large file I see log on server side that file is already uploading

```
2023-11-03 10:35:02 INFO gns3server.utils.images:325 Writing image file to '/tmp/gns3/images/debian-12-amd64.qcow2'
```

When I checked size of `/tmp/gns3/images/debian-12-amd64.qcow2.tmp` it shows `13M`

```shell
$ du -h /tmp/gns3/images/debian-12-amd64.qcow2.tmp
13M     /tmp/gns3/images/debian-12-amd64.qcow2.tmp
```

Tried upload smaller files (about 300 KB) as well and works works as expected.

**Comment by cristian-ciobanu** [comment]
I can confirm this issue happened to me many times when using GNS3 3.0.0 alpha versions. I'm using GNS3 server and GUI on the same Linux machine.

I was uploading different ISO files to install new VMs and many times the upload progress got stuck at different percents. 

After many retries I managed to complete the upload.

I guess there is a bug somewhere but it's hard to track since I did not see a clear error or some exceptions in the logs.

**Comment by ventaquil** [comment]
Same here - there are no errors in logs and server just waits for the chunk.

Maybe it's the problem with GUI?

I will try to investigate on my own.

**Comment by grossmj** [comment]
Thanks, it would be interesting to compare with the web-ui. I am going to run some tests soon.

**Comment by grossmj** [comment]
Do you use the images manager to upload?


![Image](https://github.com/GNS3/gns3-server/assets/4752614/197a8b4d-0b93-4c1e-b57c-cc1d6574a909)

**Comment by cristian-ciobanu** [comment]
I usually do not upload images through Image manager since the server and gui are on the same machine, but I simply copy the image in the corresponding sub-folder under Images/QEMU folder.
This way works also since the auto-discover process scans everything under QEMU folder periodically and after few time the image is visible in Image manager

When using Image manager it tries to upload the image directly under QEMU folder instead of allowing me to choose a subfolder.

ISO images are not allowed by default to be uploaded using Image manager, since the Filter is set to images but I tried now by removing the filter and leave it empty then I can upload ISO images but it freezes at 2%.

I also tried with some qcow2 images and two times it has frozen at 2% and 5% and the third time it worked.

The problem I experienced initially was that I was dragging a device with an empty disk on the topology then I configured it to boot from ISO and manually paste the full path to the ISO image or browse for it on the CD/DVD page then it tries to upload the ISO but freezes at a random percent and after many retries it might work.

**Comment by grossmj** [comment]
@cristian-ciobanu thanks, have you try to upload from the web-ui?

**Comment by cristian-ciobanu** [comment]
I have tried also to upload several qcow2 images using Image Manager form web-ui  and it appears to work fine.

I had only one exception, when trying to upload this qcow2 file **c8000v-universalk9_16G_serial.17.12.01a.qcow2** (using Google Chrome) every time the page crashes with a message "Something went wrong " Error code: SIGILL"

After that I tried again with the same image using Firefox at I was able to upload it.

So in general I can confirm that upload through web-ui it is working.

**Comment by cristian-ciobanu** [comment]
The problem with the image upload is still there when trying to upload an ISO image using the GUI client. 
I just tested also with the web-ui and from there is worked fine, when I checked the node properties under CD/DVD tab in the GNS3 gui I saw the ISO image name being listed in the Image field, so I assume the problem is in the desktop client.

Also i see in the debug log for the GNS3 server this message which I'm not sure it is related to this problem

`ERROR gns3server.services.authentication:74 A JWT secret key must be configured to secure the server, using an unsecured default key!`

**Comment by grossmj** [comment]
> The problem with the image upload is still there when trying to upload an ISO image using the GUI client.

Have you tried with the alpha 6? I pushed a fix but I doubt it solves the issue.

> Also i see in the debug log for the GNS3 server this message which I'm not sure it is related to this problem

Not a related issue, however we will have to address this in the setup doc.

**Comment by cristian-ciobanu** [comment]
> > The problem with the image upload is still there when trying to upload an ISO image using the GUI client.
> 
> Have you tried with the alpha 6? I pushed a fix but I doubt it solves the issue.
> 
Yes I have tried with the latest alpha 6 when I did the tests yesterday.
